ir/expression.def

```diff
@@ -404,21 +404,52 @@ class ConstructorCallExpression : Expression {
 
 /// Represents a list of expressions separated by commas
 class ListExpression : Expression {
```

Comment:
You can add a new interface IHasDefaultInitializer with a method bool hasDefaultInitializer() and have all these classes implement that interface.